psi4/src/psi4/libmints/petitelist.h

```diff
@@ -69,21 +69,11 @@ inline int64_t i_offset64(int64_t i) { return ((i * (i + 1)) >> 1); }
  *  \param mol Molecule to form mapping matrix from.
  *  \returns Integer matrix of dimension natoms X nirreps.
  */
-int **compute_atom_map(const std::shared_ptr<Molecule> &mol, double tol = 0.1, bool suppress_mol_print_in_exc = false);
-int **compute_atom_map(const Molecule *mol, double tol = 0.1, bool suppress_mol_print_in_exc = false);
+  std::vector<std::vector<int>> compute_atom_map(const std::shared_ptr<Molecule> &mol, double tol = 0.1, bool suppress_mol_print_in_exc = false);
+  std::vector<std::vector<int>> compute_atom_map(const Molecule *mol, double tol = 0.1, bool suppress_mol_print_in_exc = false);
 /// @}
 
-/*! @{
- * Frees atom mapping for created for a molecule by compute_atom_map.
- *
- *  \param atom_map Map to free.
- *  \param mol Molecule used to create atom_map.
- */
-void delete_atom_map(int **atom_map, const std::shared_ptr<Molecule> &mol);
-void delete_atom_map(int **atom_map, const Molecule *mol);
-/// @}
-
-ShellMapType compute_shell_map(int **atom_map, const std::shared_ptr<BasisSet> &);
+  ShellMapType compute_shell_map(const std::vector<std::vector<int>> & atom_map, const std::shared_ptr<BasisSet> &);
 PSI_DEPRECATED(
```

Comment:
While you're here, can you finally delete `delete_shell_map` here and in `petitelist.cc`?

Reply:
done